wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java

```diff
@@ -199,23 +208,90 @@ public void checkCredentials() throws IOException, MediaWikiApiErrorException {
 	 *
 	 * @return name of the logged in user
 	 */
+	@JsonProperty("username")
 	public String getCurrentUser() {
-		return this.username;
+		return username;
+	}
+
+	/**
+	 * Returns the map of tokens (such as csrf token and login token) currently used in this connection.
+	 */
+	public Map<String, String> getTokens() {
+		return Collections.unmodifiableMap(tokens);
+	}
+
+	/**
+	 * Sets the maximum time to wait for when establishing a connection, in milliseconds.
+	 * For negative values, no timeout is set.
+	 *
+	 * @see HttpURLConnection#setConnectTimeout
+	 */
+	public void setConnectTimeout(int timeout) {
+		connectTimeout = timeout;
+		if (timeout >= 0) {
+			buildClient();
```

Comment:
Maybe dump idea: what about setting `client = null` here? This way, the client would not be initalized multiple times if the API user set both the read and connect timeout.

Some question for the read timeout.